File: 2-add-two-numbers/add-two-numbers.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun addTwoNumbers(l1: ListNode?, l2: ListNode?): ListNode? {
        var f = l1
        var s = l2
        val res : ListNode? = ListNode(0)
        var current = res 
        var carry = 0
        while(f != null && s != null){
            val temp: Int = f.`val` + s.`val` + carry
            current!!.next = ListNode(temp%10)
            current = current!!.next
            f = f.next
            s = s.next
            carry = temp/10
        }

        while(f != null){
            val temp = f.`val` + carry
            current!!.next = ListNode(temp%10)
            current = current!!.next
            f = f.next
            carry = temp/10
        }
        while(s != null){
            val temp = s.`val` + carry
            current!!.next = ListNode(temp%10)
            current = current!!.next
            s = s.next
            carry = temp/10
        }
        if (carry == 1){
            current!!.next = ListNode(1)
        }
        return res!!.next
    }
}

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        carry = 0
        dummy = ListNode()
        curr = dummy
        while l1 or l2:
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
            val = val1+val2+carry
            carry = val//10
            val %= 10
            curr.next = ListNode(val)
            curr = curr.next
        if carry:
            curr.next = ListNode(carry)
        return dummy.next